### DISCUSSION TITLE
Теги из Last.fm

### DISCUSSION THREAD

**Discussion by vokinpirks** [Chimildic/goofy#63] [Идея] [answered]
Приветствую!
Играюсь тут с Goofy, думаю, как организовать свою библиотеку. В частности, хотелось бы разложить треки, которые я лайкнул по жанрам и другим описательным хар-кам по разным плейлистам. Жанры в самом спотифае - total mess. Они неточные, применяются на уровне артиста, а не трека/альбома. Мне больше нравится, как проставлены тэги пользователями last.fm. К тому же, я могу и сам их проставлять, что даёт мне определенный контроль. Таким образом, я мог бы написать скрипт, который бы автоматически раскладывал бы мои любимые песни по интересующим меня тегам в плейлисты, например metal, classical, jazz, idm, experimental, chill/etc. Думаю, идея понятна. Но проблема в том, что в API Goofy я не нашел как добраться до тегов ласта, ни пользовательских, ни глобальных.

**Comment by Chimildic** [comment, marked as answer]
В 1.5.4 появилась функция [rangeTag](https://chimildic.github.io/goofy/#/func?id=rangetags)

**Comment by vokinpirks** [comment]
Сейчас при работе с ластом уже и так делается N запросов к спотифаю для поиска треков. Станет "всего лишь" N * 2. Зато нормальная классификация появилась бы. Запрос к тегам, к тому же, хорошо поддается кэшированию - теги меняются редко, и, поскольку будет много повторных запросов по тегам к именно лайкнутным трекам, коих ограниченное кол-во, то будет и большой процент кэш хитов. 
А как вы определяете, что "надо всем"? Просто интересно. Гуфи не существовал всего неск. месяцев назад, но это же не значит, что он никому не нужен был. Функционал может стать востребованным после его реализации, я вот, например, не знал, что мне нужен Spotify Connect, пока про него не узнал. Примеров масса. Лично мне неинтересны большинство предлагаемых шаблонов, особенно те, где идет сбор треков с чужих плейлистов или с рекомендаций спотифая, поскольку их качество околонулевое.   
Но я вовсе не уговариваю реализовать это, думаю, в принципе, и сам смогу допилить, как мне надо. 
Спасибо.

**Comment by amd64fox** [comment]
Согласен, что работать с тегами spotify невозможно, уверен тут ситуация классическая для api spotify, народу дали ограниченный список тегов, а сами юзают полный. 
 А что если например создать свой тег файл в папке goofy по всем артистам last fm, а потом его использовать как фильтр ?
Для исполнителя ласт хранит в среднем 4-5 тегов, даже если использовать только эти теги то точность сортировки возрастёт в разы.
